scripts/modules/task-manager/scope-adjustment.js

```diff
@@ -0,0 +1,828 @@
+- Simplify implementation steps
+- Remove non-essential features
+- Streamline to basic requirements`
+				: `- Focus mainly on core functionality
+- Slightly simplify implementation steps
+- Remove some non-essential features
+- Streamline most requirements`
+}
+
+Return a JSON object with a "subtasks" array. Each subtask should have:
+- id: Sequential number starting from 1
+- title: Clear, specific title
+- description: Detailed description
+- dependencies: Array of dependency IDs as STRINGS (use format ["${task.id}.1", "${task.id}.2"] for siblings, or empty array [] for no dependencies)
+- details: Implementation guidance
+- status: "pending"
+- testStrategy: Testing approach
+
+IMPORTANT: Dependencies must be strings, not numbers!
+
+Ensure the JSON is valid and properly formatted.`;
+
+		// Define subtask schema
+		const subtaskSchema = z.object({
+			subtasks: z.array(
+				z.object({
+					id: z.number().int().positive(),
+					title: z.string().min(5),
+					description: z.string().min(10),
+					dependencies: z.array(z.string()),
+					details: z.string().min(20),
+					status: z.string().default('pending'),
+					testStrategy: z.string().nullable().default('')
+				})
+			)
+		});
+
+		const aiResult = await generateObjectService({
+			role: 'main',
+			session: context.session,
+			systemPrompt,
+			prompt,
+			schema: subtaskSchema,
+			objectName: 'subtask_regeneration',
+			commandName: context.commandName || `subtask-regen-${direction}`,
+			outputType: context.outputType || 'cli'
+		});
+
+		const generatedSubtasks = aiResult.mainResult.subtasks || [];
+
+		// Update task with preserved subtasks + newly generated ones
+		task.subtasks = [...preservedSubtasks, ...generatedSubtasks];
+
+		return {
+			updatedTask: task,
+			regenerated: true,
+			preserved: preservedSubtasks.length,
+			generated: generatedSubtasks.length
+		};
+	} catch (error) {
+		console.log(
+			`[WARN] Failed to regenerate subtasks for task ${task.id}: ${error.message}`
+		);
+		// Don't fail the whole operation if subtask regeneration fails
+		return {
+			updatedTask: task,
+			regenerated: false,
+			preserved: preservedSubtasks.length,
+			generated: 0,
+			error: error.message
+		};
+	}
+}
+
+/**
+ * Generates AI prompt for scope adjustment
+ * @param {Object} task - The task to adjust
+ * @param {string} direction - 'up' or 'down'
+ * @param {string} strength - 'light', 'regular', or 'heavy'
+ * @param {string} customPrompt - Optional custom instructions
+ * @returns {string} The generated prompt
+ */
+function generateScopePrompt(task, direction, strength, customPrompt) {
+	const isUp = direction === 'up';
+	const strengthDescriptions = {
+		light: isUp ? 'minor enhancements' : 'slight simplifications',
+		regular: isUp
+			? 'moderate complexity increases'
+			: 'moderate simplifications',
+		heavy: isUp ? 'significant complexity additions' : 'major simplifications'
+	};
+
+	let basePrompt = `You are tasked with adjusting the complexity of a task. 
```

Comment:
we have a place where we put prompts in src/prompts, would be nice to keep that structure, will also allow you to pollute this file less. 800 lines in a file is a lot.